src/scripts/list_pipelines.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A script to detect all pipelines declared in pipelines/ and cache/ and
+# dump a summary of their configurations within a `tmp/` folder in the
+# project root.
+
+import sys
+import os
+import json
+
+sys.path.append(os.getcwd())
+
+from pandas import DataFrame
+from lib.pipeline import DataPipeline
+from lib.utils import ROOT
+from typing import List, Iterator, Dict
+
+
+def get_pipeline_names() -> List[str]:
+    for item in (ROOT / "src" / "pipelines").iterdir():
+        if not item.name.startswith("_") and not item.is_file():
+            yield item.name
+
+
+def get_source_configs(pipeline_names: List[str]) -> Iterator[Dict]:
+    """Map a list of pipeline names to their source configs."""
+
+    for pipeline_name in pipeline_names:
+        data_pipeline = DataPipeline.load(pipeline_name)
+
+        for data_source in data_pipeline.data_sources:
+            data_source_config = data_source.config
+
+            data_source_name = data_source_config.get("name")
+            data_source_fetch_params = data_source_config.get("fetch", [])
+
+            for fetch_param in data_source_fetch_params:
+                file_ext = fetch_param.get('opts', {}).get('ext', '')
+                data_source_url = fetch_param.get("url")
+
+                yield {
```

Comment:
The column names here are slightly different from in https://docs.google.com/spreadsheets/d/1OGkPZEudKOvUMEciHktFUtmsLcQCDnco67YGlEYT2ok/edit?ts=5efa3bc1#gid=995000782 -- not sure if that is intentional.

Reply:
Yeah, I renamed them in the trix to be more human readable.